server/aws-lsp-codewhisperer/src/language-server/workspaceContext/artifactManager.ts

```diff
@@ -633,25 +637,49 @@ export class ArtifactManager {
             return files
         }
 
+        // Extract project roots from file paths for scenarios that workspace were not opened up from project roots
+        const projectRoots = this.extractJavaProjectRoots(files)
         const additionalFiles: FileMetadata[] = []
 
-        // Generate Eclipse configuration files
-        const javaManager = new JavaProjectAnalyzer(workspacePath)
-        const structure = await javaManager.analyze()
-        const generator = new EclipseConfigGenerator(workspaceFolder, this.logging)
+        // Process each project root separately
+        for (const projectRoot of projectRoots) {
+            const isRootProject = projectRoot === '.'
```

Comment:
why can it be '.'

Reply:
This is returned from the extraction methods above. If it's already within a root folder (with those build file) it will treat the projects roots array returned as ['.']  and `const projectPath = path.join(workspacePath, projectRoot)`  projectPath won't change still be `workspacePath`.